### PR TITLE
Remove custom JSONField from stats.db

### LINKED ISSUE #8535: Refactor stats.models to make use of django-extension's JSONField
We have our own JSONField implementation which is ugly and unnecessary to maintain.

django-extension does have a `JSONField` implementation that is compatible with ours, let's use that.

### PR DESCRIPTION
**Pull request by EnTeQuAk**
This removes our custom JSONField implementation from stats.db and re-uses django-extensions.

* Upgrades django-extensions to last known version. I don't recall why we pinned it, but all problems I see (Django compat, Python compat etc) do not exist anymore, even 2.0.7 supports everything we need
* Fix an old relict from when we removed pickle support from stats, this is only affecting tests

Fixes #8535

### CODE REVIEW

**Comment by diox** [comment]
FWIW I suspect we pinned that version because compatibility with Django 1.8 was removed.

**Comment by EnTeQuAk** [comment]
> FWIW I suspect we pinned that version because compatibility with Django 1.8 was removed.

Interestingly they're still testing against 1.8 looking at https://github.com/django-extensions/django-extensions/blob/master/tox.ini#L12

**Comment by diox** [comment]
Eh if it works, it works, fine :-)

File: src/olympia/stats/fixtures/stats/test_models.json [hunk 1]
```diff
@@ -188,7 +188,7 @@
             "versions": "{\"2.0\": 800, \"1.0\": 200}",
             "statuses": "{\"userEnabled\": 950, \"userDisabled\": 50}",
             "applications": "{\"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}\": {\"4.0\": 1500}}",
-            "oses": "a:2:{s:5:\"Linux\";i:400;s:5:\"WINNT\";i:500;}",
```
Comment (eviljeff) [inline src/olympia/stats/fixtures/stats/test_models.json]: 😨 what format was that supposed to be?

**Comment by EnTeQuAk** [comment]
> Seems like a sensible change... but I worry that everything currently stored in the database may not be valid JSON. StatsDictField wrapped its loads and dumps in try: ... except and JSONField doesn't appear to.

We actually went through that when I removed pickle support. Since then we only used `json.dumps` to store stuff in the database and the examples I looked at in the stats database on prod all look fine.

File: src/olympia/stats/fixtures/stats/test_models.json (hunk 1 above)
Comment (EnTeQuAk) [reply src/olympia/stats/fixtures/stats/test_models.json]: pickle :rofl:

**Pull request merged**